src/App.js

```diff
@@ -1,24 +1,44 @@
 import React from 'react';
 import { BrowserRouter as Router, Route } from 'react-router-dom';
+import { USER_TOKEN } from './constants';
+import getToken from './tokenScript';
+import NavButton from './components/navbutton';
+import Shopping from './components/shopping';
+import AddItem from './components/addItem';
+import { useToken } from './lib/useToken';
 import './App.css';
-import NavButton from './components/navbutton.js';
-import Shopping from './components/shopping.js';
-import AddItem from './components/addItem.js';
 
 function App() {
-  return (
-    <Router>
-      <div className="App">
-        <Route exact path="/" component={Shopping} />
-        <Route path="/add" component={AddItem} />
+  const [userToken, setToken] = useToken();
```

Comment:
We also created a custom hook, but we're very uncertain as to if this is really doing us any good that we couldn't have gotten from `useState`. Basically we don't know if we created it correctly and/or are using it well. We're both still working on reading/watching/understanding hooks and `useEffect`.